### DISCUSSION TITLE
Issues with `MVTLayer` & `autoHighlight` when filtering data

### DISCUSSION THREAD

**Discussion by DanielBoa** [visgl/deck.gl#7988] [Q&A] [answered]
I've got an issue which hopefully someone can point me in the right direction with, I might be doing something obviously wrong.

I'm trying to get `autoHighlight` to work on an `MVTLayer` which is passed a custom `renderSubLayers` function in order to filter/split data into separate sub layers, however it doesn't seem to work as I'd expect.

[I've created this minimal example](https://codesandbox.io/s/mvtlayer-autohighlight-issue-skr47y). Note that the `autoHighlight` seems to work fine until we start filtering the data passed to the child `GeoJsonLayer` (you can toggle using the button at the bottom of the example).

**Comment by DanielBoa** [comment]
Thread in slack https://deckgl.slack.com/archives/C34AC5MSQ/p1688739606673159

**Comment by Pessimistress** [comment, marked as answer]
The MVTLayer has its own special implementation of auto highlight to handle features that split across multiple tiles. While most layers just highlight the hovered object index, this layer must translate the hovered object index into multiple indices to highlight in different tiles/sublayers. In `binary:false` mode, it has to do so on CPU and always searches in unfiltered data (i.e. `tile.content`): https://github.com/visgl/deck.gl/blob/cc727dc7b7f42b201529d1ebb6ece81a7ce71c9d/modules/geo-layers/src/mvt-layer/mvt-layer.ts#L330

In this use case, I recommend filtering the tile data in `onTileLoad` instead of `renderSubLayers`.
